### PR TITLE
Ensure session by default

### PR DESCRIPTION
**Pull request by tekin**
This is to ensure all controllers require an authenticated session by default, unless explicitly overridden.

The first three commits in this PR reorganise the specs a little to more closely follow the organisation as described in the [rspec-rails project's README](https://github.com/rspec/rspec-rails/blob/master/README.md), which boils down to: request specs for testing controller endpoints (more traditionally known as controller specs) and feature specs for multi-request end-to-end browser-driven specs. This organisation makes it easier to write quick and simple request specs for verifying the various controller behaviours, with the feature specs reserved for verifying the main user flows through the application.

### CODE REVIEW

**Comment by tekin** [comment]
[Require login to access everything but home and tools page](https://trello.com/c/vL2XYQl4/243-require-login-to-access-everything-but-home-and-tools-page)

File: spec/requests/tasks_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    assert_select "#task-#{in_review_task_id}" do
+      assert_select 'a', text: 'View submission status'
+    end
+  end
+
+  it 'links to the template for a task’s framework' do
+    cboard5_task_id = '2d98639e-5260-411f-a5ee-61847a2e067c'
+
+    assert_select "#task-#{cboard5_task_id}" do
```
Comment (leeky) [inline spec/requests/tasks_spec.rb]: I find Capybara's syntax more readable:

```ruby
within "#task-#{cboard5_task_id}" do
  expect(page).to have_link 'Download template', href: '/templates/CBOARD5 MISO Data Template (July 2018).xls'
end
```

...and it has the benefit of keeping these and the feature specs consistent.

But happy to discuss!

File: spec/requests/submissions_spec.rb [hunk 2]
```diff
@@ -2,50 +2,59 @@
...
-  end
-
-  it 'shows completed submission page for a "completed" submission' do
-    mock_submission_completed_endpoint!
-    get task_submission_path(task_id: mock_task_id, id: mock_submission_id)
-
-    expect(response).to be_successful
-    expect(response.body).to include 'Submission completed'
+  context 'when signed-in' do
+    before { stub_signed_in_user }
```
Comment (leeky) [inline spec/requests/submissions_spec.rb]: Just me being fussy, probably, but a line break between `before` and the `it` below would be helpful here :)

File: spec/requests/tasks_spec.rb (hunk 1 above)
Comment (leeky) [reply spec/requests/tasks_spec.rb]: NB: I discussed this offline with @tekin and it looks like the Capybara syntax wasn't working as expected and the `within` block was silencing errors!

**Pull request merged**